### PR TITLE
Refactoring scene optimizer into separate files

### PR DESCRIPTION
**Pull request by ayushbaid**
As we add more code for auxiliary tasks like metrics, visualizations, saving, the class for each plate becomes bigger and bigger. 

This PR separates out the classes in the scene optimizer into one class per file.

### CODE REVIEW

File: gtsfm/two_view_estimator.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        self,
+        keypoints_i1_graph: Delayed,
+        keypoints_i2_graph: Delayed,
+        descriptors_i1_graph: Delayed,
+        descriptors_i2_graph: Delayed,
+        camera_intrinsics_i1_graph: Delayed,
+        camera_intrinsics_i2_graph: Delayed,
+        exact_intrinsics: bool = True,
+    ) -> Tuple[Delayed, Delayed, Delayed]:
+        """Create delayed tasks for matching and verification."""
```
Comment (akshay-krishnan) [inline gtsfm/two_view_estimator.py:54]: Missing arguments and return part of the docstring.

File: gtsfm/multi_view_optimizer.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+            {k: unit_translations[k] for k in selected_edges},
+        )
+
+    @classmethod
+    def init_cameras(
+        cls,
+        wRi_list: List[Optional[Rot3]],
+        wti_list: List[Optional[Unit3]],
+        intrinsics_list: List[Cal3Bundler],
+    ) -> Dict[int, PinholeCameraCal3Bundler]:
```
Comment (akshay-krishnan) [inline gtsfm/multi_view_optimizer.py:137]: Missing args and return.

File: gtsfm/multi_view_optimizer.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+    def create_computation_graph(
+        self,
+        num_images: int,
+        keypoints_graph: List[Delayed],
+        i2Ri1_graph: Dict[Tuple[int, int], Delayed],
+        i2Ui1_graph: Dict[Tuple[int, int], Delayed],
+        v_corr_idxs_graph: Dict[Tuple[int, int], Delayed],
+        intrinsics_graph: List[Delayed],
+    ) -> Tuple[Delayed, Delayed]:
```
Comment (akshay-krishnan) [inline gtsfm/multi_view_optimizer.py:63]: Missing docstring.

File: gtsfm/feature_extractor.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+class FeatureExtractor:
+    """Wrapper for running detection and description on each image."""
+
+    def __init__(self, detector_descriptor: DetectorDescriptorBase):
+        self.detector_descriptor = detector_descriptor
+
+    def create_computation_graph(
+        self, image_graph: Delayed
+    ) -> Tuple[Delayed, Delayed]:
+        """ Given an image, create detection and descriptor generation tasks """
```
Comment (akshay-krishnan) [inline gtsfm/feature_extractor.py:46]: Missing args for docstring.

**Review by akshay-krishnan — APPROVED** [review]
Looks good, Ayush. Except that the arguments and return sections of many methods in classes are missing. Please feel free to merge after that change.

**Review by johnwlambert — APPROVED** [review]

**Comment by ayushbaid** [comment]
Added missing docstring and moved functions outside of classes.

**Review by johnwlambert — APPROVED** [review]

**Pull request merged**